zipkin_opentracing/src/opentracing.cc

```diff
@@ -218,6 +218,23 @@ class OtSpan : public ot::Span {
     for (const auto &tag : tags_) {
       span_->addBinaryAnnotation(toBinaryAnnotation(tag.first, tag.second));
     }
+
+    std::vector<ot::LogRecord> log_records;
+    log_records.swap(log_records_);
+    log_records.reserve(options.log_records.size());
+    std::copy(options.log_records.begin(), options.log_records.end(),
+            std::back_inserter(log_records));
```

Comment:
I'm not sure we need to be making copies like this. Could we instead add a static function called something like `appendLogAnnotations` and then replace this with something like
```cpp
appendLogAnnotations(log_records_, span_);
appendLogAnnotations(options.log_records, span_);
```